src/mkdocstrings/handlers/__init__.py

```diff
@@ -128,10 +163,16 @@ def __init__(self, directory: str, theme: str, custom_templates: Optional[str] =
             paths.append(themes_dir / self.fallback_theme)
 
         self.env = Environment(autoescape=True, loader=FileSystemLoader(paths))  # type: ignore
-        self.env.filters["highlight"] = do_highlight
         self.env.filters["any"] = do_any
         self.env.globals["log"] = get_template_logger()
 
+        if theme == "readthedocs":
+            highlight_function = do_js_highlight
```

Comment:
readthedocs theme also seamlessly supports non-js highlight, I've been using it here
(note that _mkdocstrings_ is **not** involved in that project, just an example)

* https://oprypin.github.io/crsfml/tutorials/
* https://github.com/oprypin/crsfml/blob/5031f10351d5697e8adf8dd1bb7c781f7fdba739/tutorials/mkdocs.yml#L16

Hope that will also be supported here.

Reply:
Honestly I still don't understand how the syntax highlighting is dealt with between MkDocs' built-in themes or the Material for MkDocs theme :confused: 

But yeah ideally we would support what the theme supports!

Reply:
I'll users open issues about this. I don't have the time to investigate how syntax highlighting is done in all the different themes :confused: 

Let's bring initial support first and see what happens.